### DISCUSSION TITLE
node SSR instance allocates excessive memory for each request

### DISCUSSION THREAD

**Discussion by sadeq-qafari** [quasarframework/quasar#8880] [CLI - SSR mode]
In both quasar-cli development running in SSR mode and production instance which uses a node server, each http request causes the instance to allocate new megabytes of memory and never release them. At the end the node instance will quit with following error:

"FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory"

How can I force quasar SSR instance to release and dispose old requests data?

**Comment by cnlf** [comment]
The same problem, how to solve it? thanks

**Reply by sadeq-qafari** [reply]
The only solution i found was using pm2 to allocate a specific amount of ram to the app's process like:
`pm2 start index.js --max-memory-restart 300M`
It restarts the process when the process exceeds the specified amount of ram.
Also upgrading to the latest version of quasar might fix the problem.

**Comment by cnlf** [comment]
@sadeq-qafari okay, thank you
